hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,4 @@
+RELEASE_TYPE: patch
+
+This patch by Phillip Schanely refactors some of our internals to support future integrations
+with symbolic execution tools and fuzzers (:issue:`3086`).  There is no user-visible change (yet!).
```

Comment:
Just noting for later that there *are* now user-visible changes to the distribution and shrinking behavior of bounded floats, and we should probably rewrite this to focus on the immediately user-visible.  I suggest leaving that alone until we finalize everything else though.

Reply:
Took a stab at this, but I bet you still want to reword it a little!